src/AutoMapper/Configuration/MapperConfigurationExpression.cs

```diff
@@ -44,41 +48,62 @@ public void AddProfile(Profile profile)
 
         public void AddProfiles(params string[] assemblyNamesToScan)
-            => AddProfilesCore(assemblyNamesToScan.Select(name => Assembly.Load(new AssemblyName(name))));
+            => AddMaps(assemblyNamesToScan.Select(name => Assembly.Load(new AssemblyName(name))));
 
         public void AddProfiles(IEnumerable<Type> typesFromAssembliesContainingProfiles)
-            => AddProfilesCore(typesFromAssembliesContainingProfiles.Select(t => t.GetTypeInfo().Assembly));
+            => AddMaps(typesFromAssembliesContainingProfiles.Select(t => t.GetTypeInfo().Assembly));
 
         public void AddProfiles(params Type[] typesFromAssembliesContainingProfiles)
-            => AddProfilesCore(typesFromAssembliesContainingProfiles.Select(t => t.GetTypeInfo().Assembly));
+            => AddMaps(typesFromAssembliesContainingProfiles.Select(t => t.GetTypeInfo().Assembly));
+
+        public void AddMaps(IEnumerable<Assembly> assembliesToScan)
+            => AddMapsCore(assembliesToScan);
+
+        public void AddMaps(params Assembly[] assembliesToScan)
+            => AddMapsCore(assembliesToScan);
+
+        public void AddMaps(IEnumerable<string> assemblyNamesToScan)
+            => AddMapsCore(assemblyNamesToScan.Select(name => Assembly.Load(new AssemblyName(name))));
 
-        private void AddProfilesCore(IEnumerable<Assembly> assembliesToScan)
+        public void AddMaps(params string[] assemblyNamesToScan)
+            => AddMapsCore(assemblyNamesToScan.Select(name => Assembly.Load(new AssemblyName(name))));
+
+        public void AddMaps(IEnumerable<Type> typesFromAssembliesContainingMappingDefinitions)
+            => AddMapsCore(typesFromAssembliesContainingMappingDefinitions.Select(t => t.GetTypeInfo().Assembly));
+
+        public void AddMaps(params Type[] typesFromAssembliesContainingMappingDefinitions)
+            => AddMapsCore(typesFromAssembliesContainingMappingDefinitions.Select(t => t.GetTypeInfo().Assembly));
+
+        private void AddMapsCore(IEnumerable<Assembly> assembliesToScan)
         {
             var allTypes = assembliesToScan.Where(a => !a.IsDynamic).SelectMany(a => a.GetDefinedTypes()).ToArray();
+            var autoMapAttributeProfile = new NamedProfile("AutoMapAttributeProfile");
 
-            var profiles =
-                allTypes
-                    .Where(t => typeof(Profile).GetTypeInfo().IsAssignableFrom(t))
-                    .Where(t => !t.IsAbstract)
-                    .Select(t => t.AsType());
-
-            foreach (var profile in profiles)
+            foreach (var type in allTypes)
             {
-                AddProfile(profile);
+                if (typeof(Profile).IsAssignableFrom(type) && !type.IsAbstract)
+                {
+                    AddProfile(type.AsType());
+                }
+                foreach (var autoMapAttribute in type.GetCustomAttributes<AutoMapAttribute>())
```

Comment:
There should be only one attribute I guess, because otherwise some kind of validation is needed.

Reply:
OK, so why not assert that by using GetCustomAttribute instead?

Reply:
OK, not that way :)  But check that the same pair is not duplicated.

Reply:
I think I'd rather leave that off for now - I'm not sure that this would really happen. What could happen is someone calling `CreateMap` and using this attribute. Today that'll be a configuration exception.

Reply:
Looks like we can't define multiple attributes anyway, unless I mark the attribute as `AllowMultiple`, which I have not. And I don't think I will - it's weird to map multiple sources to the same destination.

Reply:
That was my initial thought too, AllowMultiple is false by default so why not GetAttribute.